templates/api_paginated_listers.mustache

```diff
@@ -0,0 +1,43 @@
+{{>partial_header}}
+package {{packageName}}
+
+{{#operations}}
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+{{#imports}}	"{{import}}"
+{{/imports}}
+)
+
+{{#operation}}
+{{#vendorExtensions}}
+{{#x-paginated}}
+func (r {{#structPrefix}}{{&classname}}{{/structPrefix}}{{^structPrefix}}Api{{/structPrefix}}{{operationId}}Request) ExecuteWithPagination() ({{#returnType}}{{^isArray}}{{^returnTypeIsPrimitive}}{{^isResponseFile}}*{{/isResponseFile}}{{/returnTypeIsPrimitive}}{{/isArray}}{{{.}}}, {{/returnType}}*http.Response, error) {
+	var items *returnType
+
+	pageNumber := int32(1)
+
+	
+
+	for {
+		page, _, err := r.Page(pageNumber).Execute()
+		if err != nil {
+			return nil, err
+		}
+
+		items = append(items, page.Devices...)
```

Comment:
Now that my confusion about vendor extension support in the generator is resolved...the `page.Devices...` here points to a need for an additional vendor extension (or maybe `x-paginated` should be an object?) so that we can capture:
- this operation is paginated
- the data property for this paginated endpoint is `devices`

Reply:
I'm curious if we have anything that looks like `Devices` in the model. @t0mk mentioned that there is a basename defined as `devices` (presumably inherited from the URL path and not the schema definitions)

We can add some debugging output in our template by emitting `this` or `.`.

https://openapi-generator.tech/docs/templating/#this

Reply:
To your point, the python script should be able to determine `Devices` since that response type schema is usually defined alongside the parameters that hint pagination.  It may be more complicated in cases (if any) where we define a Response schema object and `$ref` to that. But it should still be possible to dig into this with Python.

Reply:
`Device` is a value that may only make sense to the Go SDK, not to others.  I think I'm proposing something more generic here:

```yaml
... a devices endpoint
    x-paginated: true
    x-paginated-property: devices # This matches [the DeviceList.yaml schema](https://github.com/equinix-labs/metal-go/blob/main/spec/oas3.fetched/components/schemas/DeviceList.yaml)
... a gateways endpoint
    x-paginated: true
    x-paginated-property: gateways # This matches [the MetalGateway.yaml schema](https://github.com/equinix-labs/metal-go/blob/main/spec/oas3.fetched/components/schemas/MetalGatewayList.yaml#L2)
```

The `x-paginated-property` attribute references a property name that exists in the schema; it's up to the generator to convert the property name to the correct format for the given language and/or generator config.  It's possible that conversion happens in code, not in templates, but if we can't directly reference the transformed name that's used when, e.g., `model_device_list.go` is generated, we should at least be able to use the [generator-provided lambdas](https://openapi-generator.tech/docs/templating/#mustache-lambdas) to replicate that transformation.

Reply:
@ctreatma you're right, as you pointed out, the baseName is not enough to generate all the relevant paginated methods, for example, `findProjectInvitations` has the result schema `InvitationList` which has the attribute `Invitations`. That's a word that is not available in the template, I just checked by going through the `{{this}}` variable, which is emitted in the form of a Java class.

I will add the `x-paginated-property` and see if the emitted code is compilable.